src/lib/reasoners/adt_rel.ml

```diff
@@ -180,7 +239,17 @@ let seen_tester r hs env =
   try HSS.mem hs (MX.find r env.seen_testers)
   with Not_found -> false
 
+(* For a uninterpreted semantic value [r] and [h] a constructor of the form:
+     (cons (d1 ty1) ... (dn tyn))
+   this function adds the new equation to [eqs]:
+     [t = cons x1 ... xn]
+   where x1, ..., xn are fresh names of type respectively ty1, ..., tyn
+   and [t] is the term associated with the uninterpreted semantic value [r].
+
+   If the tester [(_ cons) t] hasn't been already seen, we also add it to
+   the equations [eqs].
```

Comment:
I assume that "seen" here means assumed, so maybe the following would be more clear:

> If the literal [(_ cons) t] was not already assumed, we add it to [eqs].

Reply:
Yes, `seen` means `assumed`. In fact, we can rename the field `seen_testers` to `assumed_testers`. I guess the old name comes from a previous implementation of the theory.